Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/exchangemgr/ExchangeManagerHelper.java

```diff
@@ -161,4 +175,251 @@ public EndpointType getServiceEndpointType(OrganizationType org, String serviceN
         }
         return null;
     }
+
+    // static-methods-ExchangeInfo_xml
+    public static List<OrganizationType> getOrganizationTypeBy(ExchangeInfoType exchangeInfo) {
+        return getOrganizationTypeBy(exchangeInfo, ELEMENT_NOT_REQUIRED);
+    }
+
+    public static List<OrganizationType> getOrganizationTypeBy(ExchangeInfoType exchangeInfo, boolean requiredElement) {
+        return getOrganizationTypeBy(exchangeInfo, null, requiredElement);
+    }
+
+    public static List<OrganizationType> getOrganizationTypeBy(ExchangeType exchange) {
+        return getOrganizationTypeBy(exchange, ELEMENT_NOT_REQUIRED);
+    }
+
+    public static List<OrganizationType> getOrganizationTypeBy(ExchangeType exchange, boolean requiredElement) {
+        if (null != exchange ) {
+            if(requiredElement && null == exchange.getOrganizationList()) {
+                exchange.setOrganizationList(new OrganizationListType());
+            }
+            if(null != exchange.getOrganizationList()){
+                return exchange.getOrganizationList().getOrganization();
+            }
+        }
+        return new ArrayList<>();
+    }
+
+    public static List<OrganizationType> getOrganizationTypeBy(ExchangeInfoType exchangeInfo, String exchangeName,
+        boolean requiredElement) {
+        return getOrganizationTypeBy(findExchangeTypeBy(getExchangeTypeBy(exchangeInfo, requiredElement), exchangeName),
+            requiredElement);
+    }
+
+    public static OrganizationType findOrganizationTypeBy(List<OrganizationType> organizations, String hcid) {
+        if (CollectionUtils.isEmpty(organizations)) {
+            return null;
+        }
+        if (StringUtils.isBlank(hcid)) {
```

Comment:
Please elaborate why we are returning an organization when hcid is blank

Reply:
hcid is now required and null is return when hcid-isBlank